catalog/catalog-service.proto

```diff
@@ -14,10 +14,8 @@ service CatalogService {
     rpc getDrinkById (DrinkRequest) returns (Drink);
     rpc getTopLevelCategories (TopLevelCategoriesRequest) returns (stream Category);
     rpc getDrinksFromCategory (DrinksFromCategoryRequest) returns (stream Drink);
-}
+    rpc getChildCategories(ChildCategoriesRequest) returns (stream Category);
```

Comment:
How to understand when we should use getDrinksFromCategory? We have no idea which category is a leaf.
Also I think it would be great to have getAllCategories() method, coz I believe we can get them all in a single request easily to reduce possible loading on the client side.

Reply:
>  We have no idea which category is a leaf.

but non-leaf category still can have a drinks

> getAllCategories()

like get **Tree** of categories? I guess it will add more complexity for client side, e.g. - when you gonna invalidate these categories?

Reply:
> but non-leaf category still can have a drinks
do we need this? any examples?

> like get Tree of categories? I guess it will add more complexity for client side, e.g. - when you gonna invalidate these categories?
hmm, ok, let's think about that later